Sources/MapboxNavigation/OverlayController.swift

```diff
@@ -0,0 +1,390 @@
+                }
+            case .failure:
+                NSLog("Failed to find visible features.")
+            }
+        })
+    }
+    
+    private func roadFeature(for line: Feature) -> (roadName: String?, shieldName: NSAttributedString?) {
+        var currentShieldName: NSAttributedString?, currentRoadName: String?
+        
+        if let ref = line.properties?["ref"] as? String,
+           let shield = line.properties?["shield"] as? String,
+           let reflen = line.properties?["reflen"] as? Int {
+            let textColor = roadShieldTextColor(line: line) ?? .black
+            let imageName = "\(shield)-\(reflen)"
+            currentShieldName = roadShieldAttributedText(for: ref, textColor: textColor, imageName: imageName)
+        }
+        
+        if let roadName = line.properties?["name"] as? String {
+            currentRoadName = roadName
+        }
+        
+        if let compositeShieldImage = currentShieldName, let roadName = currentRoadName {
+            let compositeShield = NSMutableAttributedString(string: " \(roadName)")
+            compositeShield.insert(compositeShieldImage, at: 0)
+            currentShieldName = compositeShield
+        }
+        
+        return (roadName: currentRoadName, shieldName: currentShieldName)
+    }
+    
+    /**
+     Method, which returns list of source identifiers, which contain streets tile set.
+     */
+    private func streetsSources() -> [StyleObjectInfo] {
+        let streetsSources = (try? navigationMapView.mapView.__map.getStyleSources().compactMap {
+            $0
+        }.filter {
+            let identifiers = tileSetIdentifiers($0.id, sourceType: $0.type)
+            return isMapboxStreets(identifiers)
+        }) ?? []
+        
+        return streetsSources
+    }
+    
+    /**
+     Method, which returns identifiers of the tile sets that make up specific source.
+     
+     This array contains multiple entries for a composited source. This property is empty for non-Mapbox-hosted tile sets and sources with type other than `vector`.
+     */
+    private func tileSetIdentifiers(_ sourceIdentifier: String, sourceType: String) -> [String] {
+        do {
+            if sourceType == "vector",
+               let properties = try navigationMapView.mapView.__map.getStyleSourceProperties(forSourceId: sourceIdentifier).value as? Dictionary<String, Any>,
+               let url = properties["url"] as? String,
+               let configurationURL = URL(string: url),
+               configurationURL.scheme == "mapbox",
+               let tileSetIdentifiers = configurationURL.host?.components(separatedBy: ",") {
+                return tileSetIdentifiers
+            }
+        } catch {
+            NSLog("Failed to get source properties with error: \(error.localizedDescription).")
+        }
+        
+        return []
+    }
+    
+    private func roadShieldTextColor(line: Feature) -> UIColor? {
```

Comment:
This method should also be part of WayNameLabel.

Reply:
Done